File: app/src/main/java/com/example/box25/OnboardingActivity.java
/*
 * Source: https://github.com/tutsplus/Android-Onboarding/blob/master/app/src/main/java/com/hathy/onboardingtutorial/OnboardingActivity.java
 * */

package com.example.box25;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

public class OnboardingActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_onboarding);
    }


    /**
     * finishes on-boarding operation by changing the user's preference ensuring that this activity
     * will not be launched the next time user starts the application.
     * Also redirects to main activity.
     */
    private void finishOnboarding() {
        SharedPreferences preferences = getSharedPreferences("my_preferences", MODE_PRIVATE);
        preferences.edit().putBoolean("onboarding_complete", true).apply();
        startActivity(new Intent(this, MainActivity.class));
    }
}

File: app/src/main/java/com/example/box25/OnBoardingActivity.kt
package com.example.box25

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_onboarding.*

/**
 *
 * Sources: https://www.youtube.com/watch?v=-wB_JE_PRTo&t=54s
 * Sources: https://github.com/ongakuer/CircleIndicator
 */
class OnBoardingActivity: AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_onboarding)


        val image = listOf(
                R.drawable.check_logo,
                R.drawable.tomato_1,
                R.drawable.mdi_directions_run
        )

        val title = listOf(
                R.string.welcome_title_1_onboarding,
                R.string.welcome_title_2_onboarding,
                R.string.welcome_title_3_onboarding
        )

        val texts = listOf(
                R.string.welcome_text_1_onboarding,
                R.string.welcome_text_2_onboarding,
                R.string.welcome_text_3_onboarding
        )

        val adapter = ViewPagerAdapter(image, title, texts)
        intro_pager_updated.adapter = adapter

        indicator.setViewPager(intro_pager_updated)
    }
}